client/src/chat.ts

```diff
@@ -305,49 +310,59 @@ function keydown(this: HTMLElement, event: JQuery.Event) {
   // https://stackoverflow.com/questions/5597060/detecting-arrow-key-presses-in-javascript
   // The tab key is only caught in the "keydown" event because it switches the input focus
   if (event.key === "ArrowUp") {
+    event.preventDefault();
     arrowUp(element);
   } else if (event.key === "ArrowDown") {
+    event.preventDefault();
     arrowDown(element);
   } else if (event.key === "Tab") {
     event.preventDefault();
     tab(element, event);
+  } else if (["Backspace", "Delete"].indexOf(event.key ?? "") !== -1) {
+    typedChatHistoryIndex = null;
   }
 }
 
-export function arrowUp(element: JQuery<HTMLElement>): void {
+function historyMatchNext(current: string, increment: number): string | null {
   if (typedChatHistoryIndex === null) {
-    typedChatHistoryIndex = 0;
-  } else {
-    typedChatHistoryIndex += 1;
-  }
-
-  // Check to see if we have reached the end of the history list
-  const finalIndex = typedChatHistory.length - 1;
-  if (typedChatHistoryIndex > finalIndex) {
-    typedChatHistoryIndex = finalIndex;
-    return;
+    if (increment < 0) {
+      return null;
+    }
+    typedChatHistoryIndex = -1;
+    typedChatHistoryPrefix = current;
+  }
+  const oldIndex = typedChatHistoryIndex;
+  while (1) {
```

Comment:
this should be rewritten without a constant condition, as per airbnb

Reply:
`for (;;)` better?

Reply:
I cannot find that section of airbnb guide.

Reply:
https://eslint.org/docs/rules/no-constant-condition

Reply:
like instead of doing an infinite loop, the code should be restructured such that the while loop contains the actual condition that the loop should be broken on

Reply:
Done. It is better w/ real loop condition.

Reply:
><